iphone/Classes/TiUITabGroup.m

```diff
@@ -370,20 +370,16 @@ - (void)tabBarController:(UITabBarController *)tabBarController didEndCustomizin
 
 - (void)setTabBarVisible_:(id)value
 {
-  BOOL visible = [TiUtils boolValue:value];
+  [self hideTabBar:[TiUtils boolValue:value] animated:NO];
+}
 
-  if ([self tabBarIsVisible] == visible) {
-    return;
+- (void)hideTabBar:(BOOL)hidden animated:(BOOL)animated
+{
+  if (@available(iOS 18.0, *)) {
+    [self.tabController setTabBarHidden:hidden animated:animated];
+  } else {
+    self.tabController.tabBar.hidden = hidden;
```

Comment:
The version prior to iOS 18 should remain as is, honoring the animation if set

Reply:
I remember it didn't work as expected on iOS 18 which is why I fixed it this way.

Reply:
I checked it again. It moves the TabBar up from the bottom edge and immediately down to the bottom-edge, making the animation look like a glitch and not hiding the TabBar.

Therefore, I created two new methods to show/hide the TabBar with animation, and made this property `tabBarVisible` to perform no animation (can be used at creation time as well which is not possible without this PR).

Reply:
But the fallback is for < iOS 18, so these users should have the exact same behavior (animated using the previous code)

Reply:
I meant that the same animation effects can be achieved by using new APIs on all iOS versions now. Checking it again for < iOS 18 as it wasn't fully functional there.

Reply:
@hansemannn Checked it further in 12.6.3.GA. I suggest that we keep the animation behaviour as is with a deprecation warning in 12.7.0 and remove in 13.0.0. There's one more issue I found that this property does not reset the window insets like `Window.tabBarHidden` does, will fix it as well.

Reply:
@hansemannn It's done now.